rust/src/mqtt/mqtt.rs

```diff
@@ -589,6 +624,64 @@ impl MQTTState {
         tx.tx_data.set_event(event as u8);
         self.transactions.push_back(tx);
     }
+
+    fn mqtt_frames(&mut self, flow: *const Flow, stream_slice: &StreamSlice, input: &MQTTMessage) {
+        let hdr = stream_slice.as_slice();
+        //MQTT payload has a fixed header of 2 bytes
+        let _mqtt_hdr = Frame::new(flow, stream_slice, hdr, 2, MQTTFrameType::Header as u8);
+        SCLogDebug!("mqtt_hdr Frame {:?}", _mqtt_hdr);
+        let rem_length = input.header.remaining_length as usize;
+        let data = &hdr[2..rem_length + 2];
+        let _mqtt_data = Frame::new(
+            flow,
+            stream_slice,
+            data,
+            rem_length as i64,
+            MQTTFrameType::Data as u8,
+        );
+        SCLogDebug!("mqtt_data Frame {:?}", _mqtt_data);
+    }
+    fn mqtt_frames_trunc(
+        &mut self, flow: *const Flow, stream_slice: &StreamSlice, input: &MQTTTruncatedData,
+        current: &[u8], msg: &MQTTMessage,
+    ) {
+        let hdr = stream_slice.as_slice();
+        let hdr_length = input.skipped_length - msg.header.remaining_length as usize;
+        let _mqtt_hdr = Frame::new(
+            flow,
+            stream_slice,
+            hdr,
+            hdr_length as i64,
+            MQTTFrameType::Header as u8,
+        );
```

Comment:
I'm curious to see what we will see in these frames.

Analizing the code some more, it seems to me that the Truncated message will have all header info, but no payload parsed by Suricata.
So, I'm thinking that maybe this is a special Frame type where folks only have access to info in the header, or, if we really wanted to give some insight into the (partial) truncated data, we could maybe use the remainder stream of the parsed message to register the data frame, and in that case the length would be mqtt max msg length.

Reply:
Now, one of the questions is what is interesting/important for us to expose to `mqtt` frames...

Reply:
so basically as far as I have understood when the length of parsed input message > max_mg_len then there are two conditions

1. Where it is greater than the current buffer meaning data exceeds the boundary of parsed as well as the current buffer. here I am trying to parse header and whatever remaining length is in current buffer I call it data
2. Where it exceeds the parsed message buffer but still lies within the boundary of the current overall buffer. here I use the skipped length to first extract the header and then remain portion of skipped length is data

Reply:
> Now, one of the questions is what is interesting/important for us to expose to `mqtt` frames...

All of this brings me back to your last comment ^

Reply:
I have left a review on the tests PR that adds some input into this convo.